plonkish_backend/src/pcs/multilinear/ipa.rs

```diff
@@ -289,25 +288,21 @@ where
         .collect::<Result<Vec<_>, _>>()?
         .into_iter()
         .multiunzip::<(Vec<_>, Vec<_>, Vec<_>)>();
-        let g_k = transcript.read_commitment()?;
-        let c = transcript.read_field_element()?;
+        let neg_c = -transcript.read_field_element()?;
 
         let xi_invs = {
             let mut xi_invs = xis.clone();
             xi_invs.iter_mut().batch_invert();
             xi_invs
         };
-        let eval_prime = xi_0 * eval;
-        let c_k = variable_base_msm(
-            chain![&xi_invs, &xis, Some(&eval_prime)],
-            chain![&ls, &rs, Some(vp.h())],
-        ) + comm.0;
         let h = MultilinearPolynomial::new(h_coeffs(&xis));
-
-        (c_k == variable_base_msm(&[c, c * h.evaluate(point) * xi_0], [&g_k, vp.h()])
-            && g_k == variable_base_msm(h.evals(), vp.g()).to_affine())
-        .then_some(())
-        .ok_or_else(|| Error::InvalidPcsOpen("Invalid multilinear IPA open".to_string()))
+        let hc = h.evals().iter().map(|h| *h * neg_c).collect_vec();
+        let u = &(xi_0 * (neg_c * h.evaluate(point) + eval));
+        let scalars = chain![&xi_invs, &xis, &hc, Some(u)];
```

Comment:
I now understand why the original `h_coeffs` looks like that! We could do this to avoid the scaling of `h`:

```suggestion
        let neg_c_h = MultilinearPolynomial::new(h_coeffs(neg_c, &xis));
        let u = &(xi_0 * (neg_c_h.evaluate(point) + eval));
        let scalars = chain![&xi_invs, &xis, neg_c_h.evals(), Some(u)];
```

And 

```diff
- fn h_coeffs<F: Field>(xi: &[F]) -> Vec<F> {
+ fn h_coeffs<F: Field>(scalar: F, xi: &[F]) -> Vec<F> {
     assert!(!xi.is_empty());
 
     let mut coeffs = vec![F::ZERO; 1 << xi.len()];
-    coeffs[0] = F::ONE;
+    coeffs[0] = scalar;
 
     for (len, xi) in xi.iter().rev().enumerate().map(|(i, xi)| (1 << i, xi)) {
         let (left, right) = coeffs.split_at_mut(len);
         let right = &mut right[0..len];
         right.copy_from_slice(left);
         parallelize(right, |(right, _)| {
             for coeff in right {
                 *coeff *= xi;
             }
         });
     }
 
     coeffs
 }
```